instance/management_endpoint_resolver_test.go

```diff
@@ -0,0 +1,163 @@
+			testAccessToken,
+			isLifecycleOperation)
+	})
+
+	It("should get the service", func() {
+		Expect(fakeCliConnection.GetServiceCallCount()).To(Equal(1))
+		Expect(fakeCliConnection.GetServiceArgsForCall(0)).To(Equal(serviceInstanceName))
+	})
+
+	Context("when the service instance is not found", func() {
+		BeforeEach(func() {
+			fakeCliConnection.GetServiceReturns(plugin_models.GetService_Model{}, testError)
+		})
+
+		It("should propagate the error", func() {
+			Expect(err).To(MatchError("service instance not found: " + errMessage))
+		})
+	})
+
+	Context("when the service instance is found", func(){
+		BeforeEach(func() {
+			fakeCliConnection.GetServiceReturns(plugin_models.GetService_Model{
+				DashboardUrl: "https://dasboard-url.some.host.name/x/y/guid",
+				Guid: "guid",
+			}, nil)
+		})
+
+		Context("when there is an error checking SCS version", func() {
+			Context("getting the API endpoint", func() {
+				BeforeEach(func() {
+					fakeCliConnection.ApiEndpointReturns("", errors.New("error getting API endpoint"))
+				})
+
+				It("should propagate the error", func() {
+					Expect(err).To(MatchError("error getting API endpoint"))
+				})
+			})
+
+			Context("getting an unexpected value for API endpoint", func() {
+				BeforeEach(func() {
+					fakeCliConnection.ApiEndpointReturns("https://apy.some.host.name", nil)
+				})
+
+				It("should propagate the error", func() {
+					Expect(err).To(MatchError("unexpected CF API endpoint: https://apy.some.host.name"))
+				})
+			})
+
+			Context("checking the V3 broker URL", func() {
+				BeforeEach(func() {
+					fakeCliConnection.ApiEndpointReturns("https://api.some.host.name", nil)
+					fakeAuthClient.DoAuthenticatedGetReturns(nil, 500, errors.New("error pinging V3 broker"))
+				})
+
+				It("should propagate the error", func() {
+					Expect(err).To(MatchError("error pinging V3 broker"))
+				})
+			})
+		})
+
+		Context("when we are in SCS3", func() {
+			BeforeEach(func() {
+				fakeCliConnection.ApiEndpointReturns("https://api.some.host.name", nil)
+				fakeAuthClient.DoAuthenticatedGetReturns(nil, 200, nil)
+			})
+
+			Context("when it's a lifecycle operation", func() {
+				BeforeEach(func() {
+					isLifecycleOperation = true
+				})
+
+				It("should return the broker URL for SCS3", func() {
+					Expect(output).To(Equal("https://scs-service-broker.some.host.name/cli/instances/guid"))
+				})
+			})
+
+			Context("when it's not a lifecycle operation", func() {
+				BeforeEach(func() {
+					isLifecycleOperation = false
+				})
+
+				It("should return the dashboard URL", func() {
+					Expect(output).To(Equal("https://dasboard-url.some.host.name/cli/instances/guid"))
+				})
+			})
+		})
+
+		Context("when we are in SCS2", func() {
+			BeforeEach(func() {
+				fakeCliConnection.ApiEndpointReturns("https://api.some.host.name", nil)
+				fakeAuthClient.DoAuthenticatedGetReturns(nil, 404, nil)
+			})
+
+			It("should return the dashboard URL", func() {
```

Comment:
It feels like we should test both the `is lifecycle operation` / `is not lifecycle operation` cases here, otherwise the behaviour is undefined for v2

Reply:
Makes sense, I skipped that because we don't make distinctions in that case, but agree that seems incomplete.